### ISSUE TITLE
Angular 12 Migration to Remove emitDecoratorMetadata Breaks Jest Config

### ISSUE THREAD

**Issue by michaelfaith** [angular/angular-cli#20823]
# 🐞 Bug report

### Command (mark with an `x`)

- [ ] new
- [ ] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [x] update
- [ ] lint
- [ ] extract-i18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

### Description

We chose to move from karma/jasmine to Jest for unit testing, and had it all setup since v10.  Part of that setup involved adding `"emitDecoratorMetadata": true` to our tsconfig.spec files.  The update schematic for going from v11 to 12 removed these emitDecoratorMetadata settings and broke our jest setup.  Easy enough to re-add them, but wanted to point it out since there was already a bug fix for a related issue (https://github.com/angular/angular-cli/commit/89360ab4876db6fcf92f5508971da49228543e08)

## 🌍 Your Environment

<pre><code>

Angular CLI: 12.0.0
Node: 14.17.0
Package Manager: npm 6.14.4
OS: win32 x64

Angular: 12.0.0
... animations, cdk, cli, common, compiler, compiler-cli, core
... forms, language-service, localize, material
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1200.0
@angular-devkit/build-angular   12.0.0
@angular-devkit/core            12.0.0
@angular-devkit/schematics      12.0.0
@angular/flex-layout            11.0.0-beta.33
@schematics/angular             11.2.11
ng-packagr                      12.0.0
rxjs                            6.6.7
typescript                      4.2.4

</code></pre>

**Comment by alan-agius4** [comment]
@michaelfaith, what is your configured test builder in angular.json?

**Comment by michaelfaith** [comment]
> @michaelfaith, what is your configured test builder in angular.json?

We're not currently using a builder for it at all.  We removed the test section.  We tried using jeb's @angular-builders/jest, but had issues with using it with Ivy that we weren't able to resolve (https://github.com/just-jeb/angular-builders/issues/680).  So I just configured it myself with jest-preset-angular, @types/jest, and jest, and have our npm test script run jest directly. My be worth trying his builder again, as it's been 9 months or so.  Or if you have another you'd suggest.
